ui/imports/shared/popups/SendModal.qml

```diff
@@ -125,171 +127,165 @@ StatusDialog {
-                            if(popup.selectedAccount) {
-                                return popup.selectedAccount.findTokenSymbolByAddress(address)
+                    }
+                    Item {
+                        width: parent.width
+                        height: amountToSendInput.height
+                        AmountInputWithCursor {
+                            id: amountToSendInput
+                            anchors.verticalCenter: parent.verticalCenter
+                            anchors.left: parent.left
+                            anchors.leftMargin: -Style.current.padding
+                            width: parent.width - assetSelector.width
+                            placeholderText: "0.00" + " " + assetSelector.selectedAsset.symbol
+                            input.edit.color: d.errorMode ? Theme.palette.dangerColor1 : Theme.palette.directColor1
+                            validators: [
+                                StatusFloatValidator{
+                                    id: floatValidator
+                                    bottom: 0
+                                    top: d.maxFiatBalance
+                                    errorMessage: ""
+                                }
+                            ]
+                            Keys.onReleased: {
+                                let amount = amountToSendInput.text.trim()
+
+                                if (!Utils.containsOnlyDigits(amount) || isNaN(amount)) {
+                                    return
+                                }
+                                if (amount === "") {
+                                    txtFiatBalance.text = "0.00"
+                                } else {
+                                    txtFiatBalance.text = popup.store.getFiatValue(amount, assetSelector.selectedAsset.symbol, popup.store.currentCurrency)
+                                }
+                                gasSelector.estimateGas()
+                                popup.recalculateRoutesAndFees()
                             }
-                            return ""
                         }
-                        onSelectedAssetChanged: {
-                            if (!assetSelector.selectedAsset) {
-                                return
+                        StatusAssetSelector {
+                            id: assetSelector
+                            anchors.verticalCenter: parent.verticalCenter
+                            anchors.right: parent.right
+                            assets: popup.selectedAccount.assets
+                            defaultToken: Style.png("tokens/DEFAULT-TOKEN@3x")
+                            getCurrencyBalanceString: function (currencyBalance) {
+                                return Utils.toLocaleString(currencyBalance.toFixed(2), popup.store.locale, {"currency": true}) + " " + popup.store.currentCurrency.toUpperCase()
```

Comment:
There are better methods to format currency; for example this one will give me errors since it doesn't know mine; it has a only a handful of hardcoded ones:

```js
var currencyFormatMap = { 
    en: "pre", 
    'en-GB': "pre", 
    'en-US': "pre", 
    it: "post", 
    fr: "post", 
    de: "post", 
    "de-DE": "post", 
    "de-AT": "prespace", 
    "de-CH": "prespace", 
    "de-LI": "post", 
    "de-BE": "post", 
    "nl": "post", 
    "nl-BE": "post", 
    "nl-NL": "post", 
    ro: "post", 
    "ro-RO": "post", 
    ru: "post", 
    "ru-RU": "post", 
    hu: "post", 
    "hu-HU": "post", 
    "da-DK": "post", 
    "nb-NO": "post" 
}; 
```

Reply:
Can you tell me which ones better in case you have already used it @caybro ?

Reply:
NVM, see https://github.com/status-im/status-desktop/pull/7258 ;) But I'll need your help with this SendModal, it's giving me headaches to get the numbers right :D